source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -871,6 +881,42 @@ bool JsonTranscoderFilter::maybeConvertGrpcStatus(Grpc::Status::GrpcStatus grpc_
   return true;
 }
 
+bool JsonTranscoderFilter::checkIfDecoderBufferLimitReached(uint64_t buffer_length) {
+  if (buffer_length > decoder_callbacks_->decoderBufferLimit()) {
```

Comment:
Not,  this is not what I read from this [flow control](https://github.com/envoyproxy/envoy/blob/main/source/docs/flow_control.md#decoder-filters)

Reply:
Yes, this is the doc I referenced. I suggest starting with [this section](https://github.com/envoyproxy/envoy/blob/main/source/docs/flow_control.md#http1-and-http2-filters).

The main point is:
> If a buffer is a streaming buffer, i.e. the buffered data will resolve over time, it should return FilterDataStatus::StopIterationAndWatermark to pause further data processing, which will cause the ConnectionManagerImpl to trigger watermark callbacks on behalf of the filter. If a filter can not make forward progress without the complete body, it should return FilterDataStatus::StopIterationAndBuffer.

As mentioned in the PR description, the transcoder cannot make forward progress without the complete body:
> Flow control via watermarking will not help the filter: Data stuck in a buffer will never be transcoded until more data comes in. If watermarking occurs, then the data will be stuck forever.

Reply:
Hence why I did not implement the watermarking. Perhaps someone else can confirm this?

Reply:
I see your points now.  So if we exceeded the buffer_limit, we have to bail out, waiting is not going to help.

Reply:
You may need to do something to generate an error reply when buffered data exceeds the allowed limit.  StopIterationNoBuffer won't produce the appropriate error response.

Reply:
I has an offline discussion with @antoniovicente, should be resolved. FYI we call `decoder_callbacks_->sendLocalReply` just a few lines below to generate the error reply.